sources/api/migration/migrations/v1.18.1/certdog-v0-1-1/src/main.rs

```diff
@@ -0,0 +1,25 @@
+use migration_helpers::common_migrations::{
+    AddPrefixesMigration, ListReplacement, ReplaceListsMigration,
+};
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// Add settings for the new certdog-toml config file
+fn run() -> Result<()> {
+    migrate(AddPrefixesMigration(vec![
+        "services.pki",
+        "configuration-files.certdog-toml",
+    ]))?;
+    migrate(ReplaceListsMigration(vec![ListReplacement {
+        setting: "services.pki.configuration-files",
+        old_vals: &[],
+        new_vals: &["certdog-toml"],
+    }]))
```

Comment:
This migration will need to be split up to fix the multiple migrate calls.

But in case it helps avoid the other extreme (more migrations than strictly necessary) - some of the migration helpers, like `ReplaceListMigrations` here, take a list of inputs rather than only one. That helps to avoid some duplication - you can do all migrations of a specific kind in the same binary, but not two different kinds.

Reply:
Yikes, nice catch. I had forgotten about this limitation.